### DISCUSSION TITLE
gcs_to_gcs operator apparently using deprecated parameter

### DISCUSSION THREAD

**Discussion by tpcbittencourt** [apache/airflow#20724] [Q&A]
## Apache Airflow version
v2.0.1

## What happened
I started to use the `GCSToGCSOperator` and, since then, I'm getting this _DeprecationWarning: The google_cloud_storage_conn_id parameter has been deprecated. You should pass the gcp_conn_id parameter._

This seems weird, because I'm not using the `google_cloud_storage_conn_id` parameter. I'm configuring the Operator as follows:

```
GCSToGCSOperator(
        task_id="move_files",
        source_bucket=default_args["bucket_name"],
        source_object=".../*.csv",
        destination_bucket=default_args["bucket_name"],
        destination_object="...",
        move_object=True,
        gcp_conn_id="connection_name",
    )
```

I suspect that the Airflow is ignoring the `gcp_conn_id` parameter and utilizing some default connection in `google_cloud_storage_conn_id`

## What you expected to happen
The operator should use the `gcp_conn_id` parameter and the DeprecationWarning shouldn't exist.

## Anything else
1. Is my suspicion correct? In other words, is `gcp_conn_id` being ignored?
2. If the answer for the question above is yes. Should I use the connection I want inside the `google_cloud_storage_conn_id` parameter?
3. It seems like the problem I'm facing is similar to this [one](https://github.com/apache/airflow/issues/10288). If yes, should I open an issue?

[x] I agree to follow this project's Code of Conduct

**Comment by josh-fell** [comment]
What version of the Google provider are you using?

**Reply by tpcbittencourt** [reply]
apache-airflow-providers-google==1.0.0

**Reply by potiuk** [reply]
That's an extremely old version. Google provider is at 6.3.0 version

**Reply by tpcbittencourt** [reply]
@potiuk Can I update the provider without updating the Airflow (it's on v2.0.1)?
Can some operators fail into production if I update only the provider?

**Reply by potiuk** [reply]
Here you have documentation on upgrades

https://airflow.apache.org/docs/apache-airflow/stable/installation/installing-from-pypi.html#installation-and-upgrade-scenarios. 

And you have changelog here https://airflow.apache.org/docs/apache-airflow-providers-google/stable/index.html

Basically you can upgrade but with 2.0.1 there are limits because most providers are long time ago 2.1+ only (you will see in the changelog more details). . I generally recommend you to upgrade to latest version of Airflow and providers. 

And yes there might breaking changes. There are were a number of breaking changes  (this is why Google is version 6 - means that there were 6 releases with breaking changes (we are following SemVer https://semver.org/.

**Reply by josh-fell** [reply]
FWIW, and to continue with @potiuk's remarks, if you can't afford to substantially upgrade the provider or Airflow versions (although highly recommended that you do as @potiuk points out nicely) at least initially, v2.0.0 of the Google provider should have solved this issue. The use of `google_cloud_storage_conn_id` wasn't deprecated from the `GCSToGCSOperator` until #13851 almost a year ago.

**Reply by tpcbittencourt** [reply]
Thank you @potiuk @josh-fell for your answers and clarifications!
For now, we will create our own operator using the GCSToGCSOperator source code. In parallel, we will try to update the Google provider as soon as we can.
